packages/compass-schema-validation/src/components/document-preview/document-preview.jsx

```diff
@@ -60,6 +67,12 @@ class DocumentPreview extends Component {
           <LoadingOverlay />
         ) : this.props.document ? (
           <Document doc={this.props.document} editable={false} />
+        ) : this.props.document === null ? (
```

Comment:
Not that this is unheard of, but kinda confusing to me that `undefined` and `null` are the way we differentiate between no documents and initial state, I would suggest changing `isLoaded` to actually represent possible loading states, like `initial | loading | success | error`, or at least store the whole array of results we get back from the aggregation instead of a document, we do limit by 1 document, but at least checking whether or not we got any results back would be more straightforward (but I would really suggest considering the first option I mentioned)

Reply:
That would indeed be much better. I had a similar feeling about it being confusing but I was not super sure, thanks for confirming. Changes are underway.